### ISSUE TITLE
Please add “Show others” in addition to “Enable live”

### ISSUE THREAD

**Issue by mirabilos** [cgeo/cgeo#7277]
Severity: wishlist

For those of us with very limited mobile traffic, it would be nice to have another menu point in the map “Show others”.

Currently, if I want to see the caches around me I have saved offline, I have two options:

① Select a list (or worse “All Caches”, which is 1200+ caches), then select the map icon, then wait for several minutes until it loaded, then jump to current location and zoom in 10 times ⇒ too slow to be usable

② Select a cache, select the navigation icon to show the map, zoom out sufficiently (all rather quick), hit “Enable live” in the menu immediately followed by “Disable live”

The second option is **massively** faster (mostly because it doesn’t load and render 1200+ caches, but only the 60 or so in my surroundings). It however does cost internet traffic.

If there were a new option “Show others” that would just load the ones stored offline, that would help. It does not even need to be a toggle, just hit it to add all saved caches to the current rectangle of the map view (plus a bit around it, like Enable Live does).

**Comment by SammysHP** [comment]
You can open the live map and select "disable live" from the menu. The option is persistent for the current session.

**Comment by mirabilos** [comment]
Yes, I know… but I want to have a menu point that… hm, let me explain from a different angle.

Imagine if I navigate to one cache, and zoom out a bit (say 500m in the scale bar). Then I press “Enable Live” from the menu. This does **two** things:
① display all saved-offline caches in the range
② do online searches for new caches and display them

What I wish is for an option that, on request, only does 1 without doing 2.

**Comment by moving-bits** [comment]
I'm not sure if I understand you right:

> Currently, if I want to see the caches around me I have saved offline, I have two options:

If you have GPS enabled you can just press the "live map" button on c:geo's main screen and it will show a map at your current location with all the caches you have saved offline.

When you do this the first time after starting c:geo, live map will be enabled, but you can disable this directly, and, as SammysHP stated, c:geo will remember this setting for the session.

Also, if you adjust the zoom setting, it will show more or less of your offline caches, depending on whether you zoom in or zoom out.

**Comment by moving-bits** [comment]
> Yes, I know… but I want to have a menu point that… hm, let me explain from a different angle.
> 
> Imagine if I navigate to one cache, and zoom out a bit (say 500m in the scale bar). Then I press “Enable Live” from the menu. This does **two** things:

If you use this way just to set a starting point for the map to display your offline caches, there is a workaround:
- open the cache you want to use as center point and open the map (without "live map" option)
- close the map and go to c:geo's main screen
- push the "live map" button (without enabling "live map")

c:geo will remember the last map position from the cache you have opened first, and will open the "live" map at the same position

**Comment by moving-bits** [comment]
Beside the workaround mentioned above we have a new feature in the pipeline (see PR #7228) which will let you do the following:
- use the search function from the main screen, using an address
- in the resulting list (normally one address only) you will get a "map" icon, which will let you jump to the map at this coordinates directly, without doing any further online search or loading caches etc.

**Comment by mirabilos** [comment]
Oleg Kainov dixit:

>Seems there is some misunderstanding here. For me I'm mostly always
>with "disable live" option and the live option *does not* do anything
>in regards of (1). So I always see my offline stored caches and I

Yes, but that’s too slow, it tries to render 1200+ caches.

The “open up only one cache, then use Enable Live to show the other
caches in the displayed area” is fast enough (with v3 Mapsforge).

**Comment by okainov** [comment]
If you have stored offline 1200 caches in the area - isn't it somehow expected that app will be a bit slow due to huge density? 
With "disable live" it shows only.stored caches, not the unsaved online ones

**Comment by mirabilos** [comment]
Oleg Kainov dixit:

>If you have stored offline 1200 caches in the area - isn't it somehow
>expected that app will be a bit slow due to huge density?

Somewhat. The 1200+ caches are over an area of 15x10 km or so.
If I view in the map only an 1x1 km area, it does not need to
try to render them all.

The currently existing “Enable live” functionality already does
that, in a way that makes things work decently fast (with the
v3 mapsforge) on the 1 GHz device. It just does too much ;-) for
that use case, anyway. It’s good for the use case it was designed
for (which I also use occasionally) but I have a new use case,
which it can do, with minimal change.

**Comment by githubber2222** [comment]
Really? It's 2019.... 1200+ caches is a very low number. Why would that render slowly?
Was just playing with a few other apps and they can handle way larger numbers.
Fastest so far was GCDroid where I used one of my GSAK databases with about 50000(!) caches and it rendered super smoothly and loaded very fast.

**Comment by mirabilos** [comment]
githubber2222 dixit:

>Really? It's 2019.... 1200+ caches is a very low number. Why would that render slowly?

It may be 2019, but my Android device was nevertheless built in 2008.
And I thank you to take that into account.

Open Source *is*, after all, about prolonging use of devices, with custom software.

**Comment by okainov** [comment]
>Really? It's 2019.... 1200+ caches is a very low number. Why would that render slowly?
Was just playing with a few other apps and they can handle way larger numbers.
Fastest so far was GCDroid where I used one of my GSAK databases with about 50000(!) caches and it rendered super smoothly and loaded very fast.

From the one side I support you as a user - I by myself often have several thousands of caches saved and see the performance issues here.
From the other side:
I'm not from core dev, but I think not the last reason is that c:geo is one of the oldest applications (if not the oldest), so I can easily believe in having a lot of legacy code, especially in the core part (map engine and caches rendering are *the engine*). And it can be easily seen that currently here is lack of developers willing to contribute more than couple of simple PRs. I'm sure you can find issues meaning something like "refactor core" in the backlog. But current team is struggling even with migrating to the new mapsforge engine and TargetSDK, unfortunately :(

**Comment by githubber2222** [comment]
Mhhhh, not getting into a lengthy discussion but the fact that open source is about prolonging the use of devices is news to me ;-). Open or closed source can both prolong the use of a device in the same way.
That being said. I ran a test on my trusy old Galaxy S4 (only 6 years old) and I agree... cgeo seems to totally die on me with a few hundred caches. Ran the same test with GCDroid and it runs as snappy as it does on my S9 with 10k+ caches...
So while not OPEN source I still found an app that allows me to use my old S4 without issues.

Lastly, I did by no means imply that you should not use your old phone but rather stating the fact that 1k caches is nothing... even for 10 years ago...

**Comment by githubber2222** [comment]
> > Really? It's 2019.... 1200+ caches is a very low number. Why would that render slowly?
> > Was just playing with a few other apps and they can handle way larger numbers.
> > Fastest so far was GCDroid where I used one of my GSAK databases with about 50000(!) caches and it rendered super smoothly and loaded very fast.
> 
> From the one side I support you as a user - I by myself often have several thousands of caches saved and see the performance issues here.
> From the other side:
> I'm not from core dev, but I think not the last reason is that c:geo is one of the oldest applications (if not the oldest), so I can easily believe in having a lot of legacy code, especially in the core part (map engine and caches rendering are _the engine_). And it can be easily seen that currently here is lack of developers willing to contribute more than couple of simple PRs. I'm sure you can find issues meaning something like "refactor core" in the backlog. But current team is struggling even with migrating to the new mapsforge engine and TargetSDK, unfortunately :(

I can see that. Seems like the 'old' devs left and mainly newer contributors here. And I totally agree that some code is simply too old and should be rewritten.

And lastly I think one issue that cgeo struggles with is trying to be the solution for EVERYTHING. By that I mean, supporting weird random caching sites that barely list any caches. The more things you support the more issues you will have.

Indeed surprised to see that this TargetSDK26 topic is dragging out so long!

**Comment by rsudev** [comment]
@githubber2222 According to our analysis the first version of GCDroid was based on c:geo code (which would be OK, we are open source) but without attribution (which would be not OK). This is not a statement of fact but my impression from decompiling the first versions of the app.
I guess in the meanwhile it evolved quite a bit from that.

**Comment by rsudev** [comment]
@mirabilos While I emulated this function easily in past years by just having mobile data disabled all the time, I want/will implement 'Enable stored' very likely (as soon as I get to it). But probably first/only for the current mapsforge maps, as it is quite easy to implement consistently there.

**Comment by stleusc** [comment]
@rsudev I don't think your analysis is/was correct. 
I am the author of GCDroid. I can tell you with 100% certainty that GCDroid was never based off c:geo. Not even my first internal version.
GCDroid was always using the Live API, always used GSAK compatible databases in the background (you can drop in a GSAK db and open it!) and always used different maps.
It would not make much sense to start with a project that has a completely different feature set to then change everything.

Indeed, I looked at some of the design elements and liked them, I did the same with every other app out there. Not in terms of code but in the way it feels to use it.
That is what you do, you get inspired by ideas. But that is NOT using code and that does NOT require any attribution. You can get inspired by open and/or closed source applications.

That's why most email client have a common 'feel' to it. Or any other app for that matter.

That being said, there never has been a single line of c:geo code in my project. And there never will.
Will I keep watching for ideas, inspirations? Sure... And c:geo devs do the same thing.

Example: I added swipe left/right in the cache details view... 2 month later c:geo had the same...

**Comment by stleusc** [comment]
> Then I apologise, maybe I have even mixed up what was said about GCDroid with another app, its been quite some years...

No need to apologise. There are a few visual similarities which might lead to that conclusion. 
But given the fact that I always had my code protected/obfuscated with ProGuard it would be very hard to determine code similarities.

That all being said... seems like the number of contributor went down quite a bit. What's going on? Well maybe wrong place for a question like that ;-)

**Comment by murggel** [comment]
We have meanwhile implemented long click on the "Disabled live" menu to show all offline caches (without zooming). See #16360 / #16753
@SammysHP  So I guess we can close here.
